docs/DeveloperGuide.md

```diff
@@ -1,38 +1,578 @@
-# Developer Guide
+# Upcycle - Developer Guide
+
+## Table of contents
+[Acknowledgements](#acknowledgements)
+1. [Introduction](#1-introduction)
+2. [Setup the environment](#2-setup-the-environment)
+3. [Design](#3-design)\
+   3.1. [Duke](#31-duke)\
+   3.2. [Command component](#32-command-component)\
+   3.3. [User component](#33-user-component)\
+   3.4. [Item component](#34-item-component)\
+   3.5. [Transaction component](#35-transaction-component)\
+   3.6. [Ui component](#36-ui-component)\
+   3.7. [Storage component](#37-storage-component)
+4. [Implementation](#4-implementation)\
+   4.1. [User-related features](#41-user-related-features)\
+   4.2. [Item-related features](#42-item-related-features)\
+   4.3. [Transaction-related features](#43-transaction-related-features)\
+   4.4. [Help command](#44-help-command)\
+   4.5. [Exit command](#45-exit-command)
+5. [Product Scope](#5-product-scope)\
+   5.1. [Target user profile](#51-target-user-profile)\
+   5.2. [Value proposition](#52-value-proposition)
+6. [User stories](#6-user-stories)
+7. [Non-functional requirements](#7-non-functional-requirements)
+8. [Glossary](#8-glossary)
+9. [Instructions for manual testing](#9-instructions-for-manual-testing)
 
 ## Acknowledgements
 
-{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
+This project is based on the skeleton code and documents of [SE-EDU AddressBook Level-3](https://se-education.org/addressbook-level3/)
+
+## 1. Introduction
+
+...To be updated
+
+## 2. Setup the environment
+
+**Step 1**: JDK 11 and IntelliJ are recommended
+
+**Step 2**: Fork [our repo](https://github.com/AY2223S1-CS2113-W12-1/tp) and clone the fork to your local computer.
+
+**Step 3**: Set up the JDK:
+1. From the main menu, select File > Project Structure > Project Settings > Project > Project SDK
+2. If ```JDK 11``` is already defined in Intellij IDEA, select it and click ```Apply```
+3. If ```JDK 11``` is not defined, select Add SDK > Download JDK, choose the necessary JDK (JDK 11) and click Download, then click ```Apply```
+
+**Step 4**: Go to File > Open, locate and open the cloned repository.
+
+**Step 5**: Verify the setup:
```

Comment:
It might be better to name the running class differently than the iP? (change Duke to Upcycle)!